src/main/java/com/ccentral4j/CCEtcdClient.java

```diff
@@ -262,9 +262,17 @@ public void refresh() {
     }
   }
 
-  private Counter getCounter(String key) {
-    refresh();
-    key = filterKey(key);
+  private Counter getCounter(String key, String ...groups) {
+    if (groups.length > 0) {
+      StringBuilder b = new StringBuilder(filterKey(key));
+      for (String group : groups) {
+        b.append(".");
```

Comment:
would calling `getCounters` with e.g. `getCounters(key, 'my.group.1', 'my.group.2')` be supported or should we strip away possible `.` from the group-name?

Reply:
Group values are stripped from all invalid characters and whitespaces are changed into underscores. Added additional test for this as well.